### PR TITLE
Make resource gatherer work for Kubemark

### PR DESCRIPTION
**Pull request by gmarek**

### CODE REVIEW

**Comment by gmarek** [comment]
@k8s-bot unit test this issue: #23879

File: test/e2e/framework/resource_usage_gatherer.go [hunk 1]
```diff
@@ -135,17 +136,27 @@ type resourceGatherWorker struct {
...
-	if err != nil {
-		Logf("Error while reading data from %v: %v", w.nodeName, err)
-		return
-	}
-	for k, v := range nodeUsage {
-		data[w.containerIDToNameMap[k]] = v
+	var data ResourceUsagePerContainer
+	if w.inKubemark {
+		data = getKubemarkMasterComponentsResourceUsage()
+		Logf("Got data: %#v", data)
```
Comment (wojtek-t) [inline test/e2e/framework/resource_usage_gatherer.go]: please remove

File: test/e2e/framework/resource_usage_gatherer.go [hunk 2]
```diff
@@ -171,6 +182,30 @@ func (w *resourceGatherWorker) gather(initialSleep time.Duration) {
...
 }
 
+func getKubemarkMasterComponentsResourceUsage() ResourceUsagePerContainer {
+	result := make(ResourceUsagePerContainer)
+	sshResult, err := SSH("ps ax -o %cpu,rss,command | tail -n +2 | grep kube", GetMasterHost()+":22", TestContext.Provider)
+	if err != nil {
+		Logf("Error when trying to SSH to master machine. Skipping probe")
+		return nil
+	}
+	Logf("Got raw result: %v", sshResult.Stdout)
```
Comment (wojtek-t) [inline test/e2e/framework/resource_usage_gatherer.go]: please remove

File: test/e2e/framework/resource_usage_gatherer.go [hunk 3]
```diff
@@ -171,6 +182,30 @@ func (w *resourceGatherWorker) gather(initialSleep time.Duration) {
...
+	scanner := bufio.NewScanner(strings.NewReader(sshResult.Stdout))
+	for scanner.Scan() {
+		var cpu float64
+		var mem uint64
+		var name string
+		fmt.Sscanf(strings.TrimSpace(scanner.Text()), "%f %d kubernetes/server/bin/%s", &cpu, &mem, &name)
+		Logf("Name: %v, CPU: %v, Mem: %v", name, cpu, mem)
```
Comment (wojtek-t) [inline test/e2e/framework/resource_usage_gatherer.go]: please remove

**Comment by wojtek-t** [comment]
Three minor comments - feel free to self-apply lgtm label after applying

File: test/e2e/framework/resource_usage_gatherer.go (hunk 1 above)
Comment (gmarek) [inline test/e2e/framework/resource_usage_gatherer.go]: Done.

File: test/e2e/framework/resource_usage_gatherer.go (hunk 2 above)
Comment (gmarek) [inline test/e2e/framework/resource_usage_gatherer.go]: Done.

File: test/e2e/framework/resource_usage_gatherer.go (hunk 3 above)
Comment (gmarek) [inline test/e2e/framework/resource_usage_gatherer.go]: Done.

**Comment by gmarek** [comment]
Applied comments. Adding a label.

**Comment by gmarek** [comment]
@k8s-bot test this issue: #23916

**Comment by gmarek** [comment]
@k8s-bot unit test this issue: #24528

**Comment by gmarek** [comment]
@k8s-bot unit test this issue: #24661

**Comment by gmarek** [comment]
@k8s-bot test this issue: #23536

**Pull request merged**